libraries/app.rb

```diff
@@ -80,23 +80,43 @@ def install_app
         end
 
         def install_dmg
+          temp_pkg_dir = ::File.join(Chef::Config[:file_cache_path], "Xcode_#{new_resource.version}")
+          temp_pkg_file = ::File.join(temp_pkg_dir, ::File.basename(new_resource.url))
+
+          directory temp_pkg_dir do
+            recursive true
+          end
+
           dmg_package new_resource.app do
             source new_resource.url
             checksum new_resource.checksum
             dmg_name ::File.basename(new_resource.url, '.dmg')
             owner 'root'
             type 'app'
+            destination temp_pkg_dir
             action :install
           end
+
+            execute "mv -f #{temp_pkg_dir}/Xcode.app #{install_dir}" do
+              only_if { Dir.exist?("#{temp_pkg_dir}/Xcode.app") }
+            end
+
+#          if new_resource.multi_install || !new_resource.install_root.eql?('/Applications')
```

Comment:
This did not work and resulted in 
```
No such file or directory @ rb_file_s_rename - (/var/chef/cache/Xcode_7_3_1/Xcode.app, /Applications/Xcode_7_3_1.app)
```
Wonder if that's the problem witnessed? Could be an ordering problem because the `dmg_package` hadn't even run when it attempted the rename, as opposed to using `only_if`